common/djangoapps/third_party_auth/utils.py

```diff
@@ -0,0 +1,43 @@
+from random import randint
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+
+from django.contrib.auth.models import User
+
+
+class GeneratorUsername(object):
+
+    def __init__(self):
+        self.separator_character = '_'
+        self.in_lowercase = True
+        self.random = True
+
+    def separator(self, fullname):
+        username = fullname.replace(' ', self.separator_character)
```

Comment:
By default, when there is no setting we are going to apply `fullname.replace(' ','' )`